pkg/oam/util/helper.go

```diff
@@ -230,35 +238,23 @@ func PassLabelAndAnnotation(parentObj oam.Object, childObj labelAnnotationObject
 // the format of the definition of a resource is <kind plurals>.<group>
 // Now the definition name of a resource could also be defined as `definition.oam.dev/name` in `metadata.annotations`
 // typeLabel specified which Definition it is, if specified, will directly get definition from label.
-func GetDefinitionName(u *unstructured.Unstructured, typeLabel string) string {
+func GetDefinitionName(dm discoverymapper.DiscoveryMapper, u *unstructured.Unstructured, typeLabel string) (string, error) {
```

Comment:
I know it's kinda late to change but it seems that it is better to make the dm a public variable in the package inside the init() function? In this way, we don't need to pass this around, all the caller of the utility functions shouldn't need to create a DiscoveryMapper. This is a pubic utility function that is actually used by other repros, this can break cross repo dependency.